src/NServiceBus.Core/OpenTelemetry/Metrics/MessagingMetricsFeature.cs

```diff
@@ -20,5 +24,26 @@ protected internal override void Setup(FeatureConfigurationContext context)
             performanceDiagnosticsBehavior,
             "Provides OpenTelemetry counters for message processing"
         );
+
+        context.Pipeline.OnReceivePipelineCompleted((e, _) =>
+        {
+            e.ProcessedMessage.Headers.TryGetValue(Headers.EnclosedMessageTypes, out var messageTypes);
+
+            var tags = new TagList(new KeyValuePair<string, object>[]
+            {
+                    new(MeterTags.EndpointDiscriminator, discriminator ?? ""),
+                    new(MeterTags.QueueName, queueNameBase ?? ""),
+                    new(MeterTags.MessageType, messageTypes ?? "")
+            });
+
+            Meters.ProcessingTime.Record((e.CompletedAt - e.StartedAt).TotalMilliseconds, tags);
```

Comment:
@bbrandt I think you're referring to this right? From what I understand is that they recommend to add a suffix to identify unit of measurement. I'm not sure if this value can be a double/float. We likely still want to report duration in floating point when reporting in seconds. The current code also passed double (result of `TotalMilliseconds`).

Reply:
btw, for histograms using seconds likely wouldn't make much sense for message processing. Wouldn't you want to have millisecond granularity at the base of a histogram?

Reply:
> btw, for histograms using seconds likely wouldn't make much sense for message processing. Wouldn't you want to have millisecond granularity at the base of a histogram?

This is addressed in comments for https://github.com/open-telemetry/opentelemetry-dotnet/issues/4797.
If you create a histogram with ms units, the default histogram boundaries to { 0, 5, 10, 25, 50, 75,
100, 250, 500, 750, 1000, 2500, 5000, 7500, 10000} and if you create with seconds as the unit the default histogram boundaries are those same values, divided by 1000.

Reply:
> @bbrandt I think you're referring to this right? From what I understand is that they recommend to add a suffix to identify unit of measurement. I'm not sure if this value can be a double/float. We likely still want to report duration in floating point when reporting in seconds. The current code also passed double (result of `TotalMilliseconds`).

Yes.  Value will be reported as a double.  

My preference is using https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.stopwatch?view=net-8.0 instead of DateTime.UtcNow, but this is a minor difference.